src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -847,4 +847,51 @@ private void testNullHeadGetCigarBAM() {
 
         testNullHeaderCigar(bamRec);
     }
+
+    @Test
+    public void testSetHeaderStrictValid() {
+        SAMRecord sam = createTestRecordHelper();
+        final SAMFileHeader samHeader = sam.getHeader();
+        Integer originalRefIndex = sam.getReferenceIndex();
+        Assert.assertTrue(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX != originalRefIndex);
+
+        // force re-resolution of the reference name
+        sam.setHeaderStrict(samHeader);
+        Assert.assertEquals(sam.getReferenceIndex(), originalRefIndex);
+    }
+
+    @Test(expectedExceptions=IllegalArgumentException.class)
+    public void testSetHeaderStrictInvalidReference() {
+        SAMRecord sam = createTestRecordHelper();
+        final SAMFileHeader samHeader = sam.getHeader();
+
+        sam.setReferenceName("unresolveable");
+        int originalRefIndex = sam.getReferenceIndex();
```

Comment:
This is odd -- didn't `setReferenceName()` clear the reference index? (and so "originalRefIndex" is not actually the "original" reference index)